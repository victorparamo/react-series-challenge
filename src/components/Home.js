import React, { Component }  from "react";
import { connect } from 'react-redux';
import { GiftContainer }  from '../styled_component/Home.styled';
import SearchBarHome from './SearchBars';
import giftActions from '../redux/actions/gifts';
import favoriteActions from '../redux/actions/favorites';
import { bindActionCreators } from 'redux';
import { Portal, Modal } from './Portals/index';



class Home extends Component {
  constructor(props){
    super(props);
    this.state = { open: false };
  }

  showModal = () => {
    this.setState((prevState) => {
      return { open: !prevState.open };
    });
  }
  
  componentDidMount(){
    this.props.getGift({ text: "" });
  }

  render(){
    const { open, } = this.state;
    const { gifts } = this.props;

    return (
      <GiftContainer>
        <SearchBarHome />
        {gifts && <GiftList
                    gifts={gifts}/>}
        {/*<ShowPortal 
          onClick={this.showModal}
        open={open}/>*/}
      </GiftContainer>
    );
  }
}

const GiftList = (props) => {
  const { gifts } = props

  return Object.values(gifts).map((item, index)=> (
    <GiftItem 
      url={item.images.preview_gif.url}
      name={item.title}/>
  ));
}

const handleFavorite = (props) => {
  const {
    addFavorite,
    gift
  } = props
  console.log("--->", props)
  addFavorite(gift);
}

const GiftItem = (props) => {
  const {url, name, addFavorite}=props
  const gift = { url, name }
  return (
    <img src={props.url} width={"100px"} height={"100px"} onClick={()=>addFavorite({gift})}/>
  );
}

const ShowPortal = ({onclick, open}) => {
  return (
    <Portal>
      <Modal open={open} onClick={onclick}>
        <span>hola</span>
      </Modal>
    </Portal>
  );
}

function mapStateToProps (state) {
  const { gifts,  } = state;
  
  return { gifts, };
};
function mapDispatchToProps (dispatch) {
  const { getGift } = giftActions.creators;
  const { addFavorite, } = favoriteActions.creators;

  return bindActionCreators({ getGift, addFavorite, }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);