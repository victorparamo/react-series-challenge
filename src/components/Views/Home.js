import React, { Component }  from "react";
import { connect } from 'react-redux';
import { SearchBarHome } from '../SearchBars';
import giftActions from '../../redux/actions/gifts';
import { bindActionCreators } from 'redux';
import GiftContainer from '../GiftContainer';
import GiftItem from '../giftItem';

class Home extends Component {
  componentDidMount(){
    this.props.getGift({ text: "" });
  }

  render(){
    const { gifts } = this.props;

    return (
        <GiftContainer
          searchBar = {<SearchBarHome />}
          gifts = {gifts}
          giftList = {gifts ? <GiftList 
                          gifts={gifts}/>:""}
          />
    );
  }
}

const GiftList = (props) => {
  const { gifts } = props

  console.log("gifts:",gifts)
  if(gifts.length < 5){
    throw new Error("Not Found")
  }
  
  return (Object.values(gifts).map((item, index)=> (
    <GiftItem 
      url={item.images.preview_gif.url}
      name={item.title}
      id={item.id}
      key={index}/>
  )));
  
};

function mapStateToProps (state) {
  const { gifts, } = state;
  
  return { gifts, };
};

function mapDispatchToProps (dispatch) {
  const { getGift } = giftActions.creators;

  return bindActionCreators({ getGift, }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);