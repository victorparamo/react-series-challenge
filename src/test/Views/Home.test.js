import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import Home from '../../components/Views/Home';
import toJson from 'enzyme-to-json';

describe('Home', () => {
    const initialState = {};

    const mockStore = configureStore();

    let store, container;

    beforeEach(()=>{
        store=mockStore(initialState)
        container=mount(<Home 
                            store={store}
                            gifts={{"cdcd":"cdcdc"}}/>)
    })

    it('Home', () => {
        expect(toJson(container)).toMatchSnapshot();
    })
})
