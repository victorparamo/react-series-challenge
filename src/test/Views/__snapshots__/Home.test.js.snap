// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home 1`] = `undefined`;
